src/test/java/seedu/address/logic/commands/GenerateLocationsCommandTest.java

```diff
@@ -107,4 +110,56 @@ public void execute_validInput_success() {
         assertCommandSuccess(command, model, expectedCommandResult, expectedModelForGenerate);
     }
+
+    @Test
+    public void execute_validInputFromViewingAllInfected_success() {
+        CommandResult expectedCommand = new CommandResult("Generated locations for: Daniel Meier",
+                false, false, CommandResult.SWITCH_TO_VIEW_LOCATIONS);
+        Model modelForAllInfected = model;
+        modelForAllInfected.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        Model expectedModelForGenerate = expectedModel;
+        Predicate<Location> locationPredicate = location -> location.getId().getOneBased() == 6;
+        expectedModelForGenerate.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        expectedModelForGenerate.updateFilteredLocationList(locationPredicate);
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        assertCommandSuccess(command, modelForAllInfected, expectedCommand, expectedModelForGenerate);
+    }
+
+    @Test
+    public void execute_invalidInputFromViewingAllQuarantined_throwCommandException() {
+        Model modelForAllQuarantined = model;
+        modelForAllQuarantined.updateFilteredPersonList(PREDICATE_SHOW_ALL_QUARANTINED);
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        assertThrows(CommandException.class, () -> command.execute(modelForAllQuarantined));
+    }
+
+    @Test
+    public void execute_validInputFromViewingAllInfected_success() {
+        String expectedMessage = "Generated locations for: Daniel Meier";
+        Model modelForAllInfected = model;
+        modelForAllInfected.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        Model expectedModelForGenerate = expectedModel;
+        Predicate<Location> locationPredicate = location -> location.getId().getOneBased() == 6;
+        expectedModelForGenerate.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        expectedModelForGenerate.updateFilteredLocationList(locationPredicate);
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        assertCommandSuccess(command, modelForAllInfected, expectedMessage, expectedModelForGenerate);
+    }
+
+    @Test
+    public void execute_invalidInputFromViewingAllQuarantined_throwCommandException() {
+        String expectedMessage = MESSAGE_PERSON_IS_NOT_INFECTED;
+        Model modelForAllQuarantined = model;
+        modelForAllQuarantined.updateFilteredPersonList(PREDICATE_SHOW_ALL_QUARANTINED);
+        Index index = Index.fromOneBased(1);
+        GenerateLocationsCommand command = new GenerateLocationsCommand(index);
+        try {
+            command.execute(modelForAllQuarantined);
+        } catch (CommandException e) {
+            assertTrue(e.getMessage().equals(expectedMessage));
```

Comment:
This test is repeated and ought to be deleted.

Reply:
Removed